pkg/operation/etcdencryption/encryptionconfiguration.go

```diff
@@ -0,0 +1,218 @@
+	}
+	if ec.APIVersion != ecAPIVersion {
+		return false, fmt.Errorf("apiversion (%v) of the EncryptionConfiguration does not match the expected apiversion (%v)", ec.APIVersion, ecAPIVersion)
+	}
+	ecResouceConfigurationLenA := len(ec.Resources)
+	if ecResouceConfigurationLenA != ecResouceConfigurationLenE {
+		return false, fmt.Errorf("number of resource configurations (%v) of the EncryptionConfiguration does not match the expected number (%v)", ecResouceConfigurationLenA, ecResouceConfigurationLenE)
+	}
+	// Check resources
+	if !sets.NewString(ec.Resources[0].Resources...).Equal(sets.NewString(ecEncryptedResources...)) {
+		return false, fmt.Errorf("list of encrypted resources (%v) of the EncryptionConfiguration does not match the expected list (%v)", ec.Resources[0].Resources, ecEncryptedResources)
+	}
+	// Check encryption providers
+	ecEncryptionProviderLenA := len(ec.Resources[0].Providers)
+	if ecEncryptionProviderLenA != ecEncryptionProviderLenE {
+		return false, fmt.Errorf("number of encryption providers (%v) of the EncryptionConfiguration does not match the expected number (%v)", ecEncryptionProviderLenA, ecEncryptionProviderLenE)
+	}
+	// Encryption provider aescbc and identity in any order are ok
+	if !(((ec.Resources[0].Providers[0].Identity != nil) && (ec.Resources[0].Providers[1].AESCBC != nil)) ||
+		((ec.Resources[0].Providers[0].AESCBC != nil) && (ec.Resources[0].Providers[1].Identity != nil))) {
+		return false, fmt.Errorf("unexpected encryption providers of the EncryptionConfiguration found. Expected are exactly two providers 'identity' and 'aescbc'")
+	}
+	var aesConfig *apiserverconfigv1.AESConfiguration
+	if ec.Resources[0].Providers[0].AESCBC != nil {
+		aesConfig = ec.Resources[0].Providers[0].AESCBC
+	} else {
+		aesConfig = ec.Resources[0].Providers[1].AESCBC
+	}
+	ecEncryptionProviderAESCBCKeyLenA := len(aesConfig.Keys)
+	if ecEncryptionProviderAESCBCKeyLenA < ecEncryptionProviderAESCBCMinKeyLenE {
+		return false, fmt.Errorf("unexpected number of keys in encryption provider aescbc of the EncryptionConfiguration found. Expected are at least %v key(s)", ecEncryptionProviderAESCBCMinKeyLenE)
+	}
+	keyNameMap := make(map[string]bool)
+	for _, key := range aesConfig.Keys {
+		ok, err := isKeyConsistent(key)
+		if (err != nil) || (!ok) {
+			return false, fmt.Errorf("inconsistent key in encryption provider aescbc of the EncryptionConfiguration found")
+		}
+		_, ok = keyNameMap[key.Name]
+		if ok {
+			return false, fmt.Errorf("two or more keys (%v) with same timestamp found in encryption provider aescbc of the EncryptionConfiguration", key.Name)
+		}
+		keyNameMap[key.Name] = true
+	}
+	return true, nil
+}
+
+// Equals checks whether the provided encryption configurations are equal.
+func Equals(ec1 *apiserverconfigv1.EncryptionConfiguration, ec2 *apiserverconfigv1.EncryptionConfiguration) (bool, error) {
```

Comment:
I think a string comparison isn't the right thing to do. Also, if you compare the entire resource, you also compare thinks like `resourceVersion` or `creationTimestamp` which do not tend to be the same - a thing that you should rather do is compare the `spec`s with `apiequality.Semantic.DeepEqual`

Reply:
These resources don't have `metadata`, however, I agree with @adracus' comment.